File: BFS/#111 MInimum Depth of Binary Tree.cpp
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root==NULL) return 0;
        queue<TreeNode*>Q;
        Q.push(root);
        int depth=0;
        while(!Q.empty()){
            depth++;
            int queueCount=Q.size(); 
            for(int i=0; i<queueCount; i++){
            TreeNode* cur = Q.front();Q.pop();
             if(!cur->left && !cur->right) return depth;
            if(cur->left) Q.push(cur->left);
            if(cur->right) Q.push(cur->right);
            }
        }
        return 0;
    }
};

File: Tree/#111. Minimum Depth of Binary Tree.py
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        min_d = None
        if not root.left or not root.right:
            return 1 + max(self.minDepth(root.left), self.minDepth(root.right))
        else:
            return 1 + min(self.minDepth(root.left), self.minDepth(root.right))
            